src/bundle/Resources/public/js/scripts/core/multilevel.popup.menu.js

```diff
@@ -0,0 +1,385 @@
+
+            subitemsWithSubitems.forEach((subitem) => {
+                this.closeWithSubbranches(subitem.branchElement);
+            });
+
+            this.closeBranch(branchElement);
+        }
+
+        generateMenu(menuTree) {
+            const { triggerElement, groups, placement, fallbackPlacements, processAfterCreated: processBranchAfterCreated } = menuTree;
+
+            const branchElement = this.generateBranch(
+                {
+                    triggerElement,
+                    placement,
+                    fallbackPlacements,
+                },
+                processBranchAfterCreated,
+            );
+
+            groups.forEach((group) => {
+                const { id: groupId, items, processAfterCreated: processGroupAfterCreated } = group;
+
+                this.generateGroup({ id: groupId, branchElement }, processGroupAfterCreated);
+
+                items.forEach((item) => {
+                    const { label, sublabel, href, onClick, processAfterCreated: processItemAfterCreated } = item;
+
+                    const itemElement = this.generateItem(
+                        { label, sublabel, branchElement, groupId, href, onClick },
+                        processItemAfterCreated,
+                    );
+
+                    if (item.branch) {
+                        this.generateMenu({
+                            ...item.branch,
+                            triggerElement: itemElement,
+                        });
+                    }
+                });
+            });
+
+            return branchElement;
+        }
+
+        generateBranch(data, processAfterCreated = () => {}) {
+            const { triggerElement, placement, fallbackPlacements } = data;
+            const { branchTemplate } = this.container.dataset;
+
+            const container = doc.createElement('div');
+            const renderedItem = branchTemplate;
+
+            container.insertAdjacentHTML('beforeend', renderedItem);
+
+            const newBranchElement = container.querySelector('.ibexa-multilevel-popup-menu__branch');
+
+            processAfterCreated(newBranchElement, data);
+
+            this.initBranch(triggerElement, newBranchElement, null, placement, fallbackPlacements);
+
+            const parentBranchElement = triggerElement.closest('.ibexa-popup-menu');
+
+            triggerElement.branchElement = newBranchElement;
+            newBranchElement.itemElement = triggerElement;
+            newBranchElement.parentBranchElement = parentBranchElement;
+
+            const isTriggerMultilevelMenuItemElement = triggerElement.classList.contains('ibexa-popup-menu__item');
+
+            if (isTriggerMultilevelMenuItemElement) {
+                triggerElement.classList.add('ibexa-popup-menu__item--has-subitems');
+            }
+
+            return newBranchElement;
+        }
+
+        // eslint-disable-next-line no-unused-vars
```

Comment:
can u explain why we need this here?
Cannot we just pass instead of arguments data and this variable?

Reply:
`arguments` is better in a case e.g. this class is overwritten.